templates/stepback-qa-prompting/stepback_qa_prompting/chain.py

```diff
@@ -32,6 +32,7 @@ def retriever(query):
 few_shot_prompt = FewShotChatMessagePromptTemplate(
     example_prompt=example_prompt,
     examples=examples,
+    input_variables=[],
```

Comment:
I think it's because of BasePromptTemplate requiring it as a param. How would you like to fix? One way I think is to just add this to FewShotChatMessagePromptTemplate class:

def __init__(self, *args, **kwargs):
        if "input_variables" in kwargs:
            super().__init__(*args, **kwargs)
        else:
            kwargs['input_variables'] = []
            super().__init__(*args, **kwargs)
            
lmk what you think is best

Reply:
Let's figure out how to update the pre init code. We want the template to be able to infer the input variables on its own if none are provided.

If one provides `[]` as a value then that should mean that the template takes *no* input variables and should raise a ValueError if the template does accept variables.